src/StatsTools/reweighting.jl

```diff
@@ -394,18 +405,238 @@ function mixed_estimator_df_progress(shift::Vector, hproj::Vector, vproj::Vector
+    end
+    dτ = df.dτ_1[end]
+
+    return rayleigh_replica_estimator(op_ol_v, vec_ol_v, shift_v, 0, dτ; skip, E_r, kwargs...)
+end
+
+"""
+    rayleigh_replica_estimator_analysis(df::DataFrame; kwargs...)
+    -> (; df_rre, df_se)
+Compute the [`rayleigh_replica_estimator`](@ref) on a `DataFrame` `df` returned from [`lomc!`](@ref)
+repeatedly over a range of reweighting depths.
+
+Returns a `NamedTuple` with the fields
+* `df_rre`: `DataFrame` with reweighting depth and `rayleigh_replica_estimator` data. See example below.
+* `df_se`: `DataFrame` with [`shift_estimator`](@ref) output, one row per replica
+
+## Keyword arguments
+* `h_range`: The default is about `h_values` values from 0 to twice the estimated correlation time
+* `h_values = 100`: minimum number of reweighting depths
+* `skip = 0`: initial time steps to exclude from averaging
+* `threading = Threads.nthreads() > 1`: if `false` a progress meter is displayed
+* `shift = "shift"`: shift data corresponding to column in `df` with names `<shift>_1`, ...
+* `op_ol = "Op1"`: name of operator overlap corresponding to column in `df` with names `c1_<op_ol>_c2`, ...
+* `vec_ol = "dot"`: name of vector-vector overlap corresponding to column in `df` with names `c1_<vec_ol>_c2`, ... 
+* `Anorm = 1`: a scalar prefactor to scale the operator overlap data
+* `warn = true`: whether to log warning messages when blocking fails or denominators are small
+
+## Example
+```julia
+df, _ = lomc!(...)
+df_rre, df_se = rayleigh_replica_estimator_analysis(df; skip=5_000)
+
+using StatsPlots
+@df df_rre plot(_ -> se, :h, ribbon = (se_l, se_u), label = "⟨S⟩") # constant line and ribbon for shift estimator
+@df df_rre plot!(:h, :val, ribbon = (:val_l, :val_u), label="E_mix") # Rayleigh quotient estimator as a function of reweighting depth
+xlabel!("h")
+```
+See also: [`rayleigh_replica_estimator`](@ref), [`mixed_estimator_analysis`](@ref), [`AllOverlaps`](@ref).
+"""
+function rayleigh_replica_estimator_analysis(
+    df::DataFrame;
+    h_range=nothing,
+    h_values=100,
+    skip=0,
+    threading=Threads.nthreads() > 1,
+    shift="shift",
+    op_ol="Op1",
+    vec_ol="dot",
+    Anorm=1,
+    warn=true,
+    kwargs...
+)
+    num_reps = length(filter(startswith("dτ"), names(df)))
+    dτ = if num_reps == 1
+        df.dτ[end]
+    else
+        df.dτ_1[end]
+    end
+    shift_v = Vector[]
+    E_r = []
+    correlation_estimate = []
+    df_se = DataFrame()
+    for a in 1:num_reps
+        push!(shift_v, Vector(getproperty(df, Symbol(shift*"_$a"))))     # overwrite column name
+        se = blocking_analysis(shift_v[a]; skip)
+        push!(E_r, se.mean)
+        push!(correlation_estimate, 2^(se.k - 1))
+        push!(df_se, (; replica=a, NamedTuple(se; name=:se)...))
+    end
+    if isnothing(h_range)
+        h_range = determine_h_range(df, skip, minimum(correlation_estimate), h_values)
+    end
+    vec_ol_v = Vector[]
+    op_ol_v = Vector[]
```

Comment:
Here, and elsewhere,  you are creating vectors that don't have the complete type information. This could lead to some performance degradation in `rayleigh_replica_estimator_df_folds`.